docs/zh/guide/custom-framework-integration.mdx

```diff
@@ -0,0 +1,330 @@
+如果使用 LynxExplorer（在[此文档](guide/start/quick-start.html)中获取它）打开这个文件，你会看到一个空白的页面，以及一些错误信息。
+
+</Steps>
+
+## 将 App Bundle 运行起来
+
+我们现在开始关注这个 App Bundle 的运行效果。
+
+我们将完善两个线程的入口文件，以便 Lynx 可以正确地运行它们，并能绘制出 UI。同时，我们还会添加CSS，来为我们 UI 添加样式。下面是最终的效果：
+
+<Go
+  example="vanilla-lynx-bundle"
+  highlight="{3}"
+  defaultFile="src/index.main-thread.js"
+  defaultEntryFile="dist/vanilla.lynx.bundle"
+  img={`${window.location.origin}/lynx-examples/vanilla-lynx-bundle/assets/preview.png`}
+  entry={/(^src\/|rspack\.config\.js)/}
+/>
+
+---
+
+请跟随我们的步骤，一步一步地完成构建：
+
+<Steps>
+
+### 添加 Called-by-Native 的函数
+
+Lynx 通过调用一些 `globalThis` 上的函数来与你的应用进行交互。这些函数被称为 Called-by-Native 函数。
+
+我们将在主线程的 `renderPage` 函数中调用 Lynx 的 [Element PAPI](api/engine/element-api.html) 来绘制一些 UI。
+而在后台线程，也有一个类似的函数 `initBundle`，由于后台线程无法直接操作 UI，因此我们只是简单地打印一些信息。
+
+<Tabs
+  defaultValue="src/index.main-thread.js"
+  values={[
+    { label: '主线程', value: 'src/index.main-thread.js' },
+    { label: '后台线程', value: 'src/index.js' },
+  ]}>
+  <Tab value="src/index.main-thread.js">
+
+```jsx title="src/index.main-thread.js" {2-14}
+globalThis.renderPage = function () {
+  const page = __CreatePage('0', 0);
+  const pageId = __GetElementUniqueID(page);
+  const container = __CreateView(pageId);
+  __AddClass(container, 'container');
+  const logo = __CreateImage(pageId);
+  __SetAttribute(logo, 'src', 'https://github.com/lynx-family.png');
+  __AddClass(logo, 'logo');
+  const slogan = __CreateText(pageId);
+  __AddClass(slogan, 'slogan');
+  __SetAttribute(slogan, 'text', 'Lynx: Unlock Native for More');
+  __AppendElement(page, container);
+  __AppendElement(container, logo);
+  __AppendElement(container, slogan);
+};
+globalThis.updatePage = function () {};
+globalThis.processData = function () {};
+```
+
+  </Tab>
+  <Tab value="src/index.js">
+
+```jsx title="src/index.js" {2}
+globalThis.initBundle = () => {
+  console.log('Lynx: Unlock Native for More');
+};
+```
+
+  </Tab>
+</Tabs>
+
+### 为 Rspack 添加 CSS 支持
+
+Rspack 提供了开箱即用的 CSS 支持。你只需要配置 `experiment.css` 为 `true`，便一切就绪了。
+
+`LynxTemplatePlugin` 插件会“收集”你的 CSS 文件，并将其添加到 App Bundle 中。
+
+```js filename=rspack.config.js {5}
+const config = {
+  ...
+  experiments: {
+    ...
+    css: true,
+  },
+};
+```
+
+### 编写样式
+
+我们还需要为 UI 添加一些样式。我们已经为你写好了 CSS 文件，请在 `src` 目录下创建一个 `index.css` 文件，并将下面的内容复制进去：
+
+```css filename=src/index.css
+.container {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  height: 100vh;
+  gap: 16rpx;
+}
+.logo {
+  width: 128rpx;
+  height: 128rpx;
+  border-radius: 64rpx;
+}
+.slogan {
+  color: #333;
+  font-weight: bold;
+}
+```
+
+当然，别忘记在入口文件中引入这个 CSS 文件，请注意不要同时在两个入口文件中引入相同的 CSS 文件，否则会导致 CSS 重复。
+
+```jsx filename=src/index.main-thread.js {1}
+import './index.css';
+
+...
+```
+
+### 构建和运行
+
+现在，你可以再次执行 `npx rspack build`，然后在 `dist` 目录下看到 `vanilla.lynx.bundle` 文件。
+使用 LynxExplorer 打开这个文件，你会看到一个带有 Lynx Logo 和 Slogan 的页面。
```

Comment:
我有几个小问题
1. 为啥你选择的是使用 Vanilla JS？咱文档不是写的要教大家集成一个框架么。我会觉得这里介绍 UI 框架和 Element PAPI 的内容有点少
2. 可以让开发者直接从 Rspeedy 开始么？这是不是可以减少开发者需要理解 `tasm` 这些细节的问题

Reply:
对于第一个问题，我有思考过直接上来就接solidjs，但是最后发现篇幅会很长，我倾向于讲一个vanilla，然后直接给solidjs的example，他要是能看懂vanilla，这个complete example也不是问题

对于第二个问题，rspeedy其实不包含任何tasm这些东西（他本质上就是rsbuild），还得是template-webpack-plugin，有了这个也不需要关注tasm这些东西了，而且还能暴露最大程度的灵活性

我整体上的思路是希望在这里暴露最大程度的灵活性、减少预设和封装、使用最通用的技术、以及强调lynx的构建和web的差异很小

Reply:
Thank you for the clarification. I believe it would be beneficial to add the following information at the beginning of the file:
- Option A: Include an introduction like, "We'll start with vanilla JavaScript to understand the basics of the element PAPI. Following this, you'll learn how to integrate it with other web frameworks."
- Option B: Directly use an example with Solid.js for a more practical demonstration.